### DISCUSSION TITLE
Wanted: CPAC LTS release. --Thoughts on my CPAC journey.

### DISCUSSION THREAD

**Discussion by joshleelab** [FCP-INDI/C-PAC#2151] [General]
Dear CPAC Team,

I first started using CPAC in 2016. I liked CPACs capabilities, ambitious vision, including iteratability of pipelines and advanced analytical features. And for that I'd like to applaud your team (current and alumni).

However, it has been a constant struggle to use. Every version I've tried since 2016 has had major bugs that prevented important aspects of functionality that drew me to the project. Over the years, I've tried various new versions, but for every bug that was fixed, new ones appeared. I appreciate the immense amount of work your team has put into CPAC. And yet, I am admitting to you all now, that as of last night, I've finally given up and turned to fmriprep. I need reliability.

As I depart (at least for now), I just want to say that you need to release a stable, Long-term support version of CPAC, a release that has either fixed or removed known broken features, and does not add new features until new bugs stop being reported.

Sincerely,

Josh Lee
UC DAVIS MIND Intitute

**Comment by sgiavasis** [comment]
Hi Josh,

Thanks for your honest feedback. We appreciate all your engagement throughout the years and you've been a great power user who has provided a good deal of issue posts and feedback, which are always helpful for us.

The release and feature planning for C-PAC is governed by several stakeholders and groups, and is thus a process by committee. Combined with this is a relatively small team of engineers, and there is ongoing tension between meeting needs of collaborators and users relying on new and updated methods and features, and upholding stability. As we are grant-funded, stipulations in the grants for certain multi-year timelines also require us to meet expectations of certain features and capabilities.

In addition, our first and foremost priority has always been ensuring the calculated outputs of C-PAC never vary from version to version. When run issues show up, often in the form of many different use cases and needs (due to C-PAC's expansive configurability), we do our best to triage these against the backdrop of these other competing concerns.

C-PAC has always been an ambitious and experimental attempt at a platform with a scale of configurability so wide that it truly is a software development and project management puzzle, and topic of internal research and tinkering, in its own right. This is one of the reasons why, in the past few years, we have also built a dedicated DevOps core and infrastructure to start helping to elevate and automate the testing and stability of C-PAC against the speed and scope of expansion.

You may find it interesting that we have been in a process of designing and implementing a new underlying engine for C-PAC that will be far more modular and flexible both for users and developers; once complete this will go a long way towards ensuring ongoing stability despite any possible future pushes for aggressive growth and development of new features.

I hope this puts things into context.

Best,
Steve
